server/pull_request_handler.go

```diff
@@ -157,29 +181,45 @@ func (s *Server) handlePullRequestEvent(ctx context.Context, event *PullRequestE
 		}
 	}
 
-	s.checkPullRequestForChanges(ctx, pr)
+	changed, err := s.checkPullRequestForChanges(ctx, pr)
+	if err != nil {
+		mlog.Error("Could not check changes for PR", mlog.Err(err))
```

Comment:
Maybe we should return 400 or 500 error here

Reply:
Makes sense, I also thought to return `500` for this but didn't want to introduce behavioral change. But I'm open to change it 👍

Reply:
I'll do it. All the HTTP errors are new but I think is necessary to be aware of problems. Those errors would be reflected in the metrics so it's a good thing